File: go/count-number-of-distinct-integers-after-reverse-operations.go
func GetReverseNum(x int) int {
	reversed := 0
	for 0 < x {
		reversed = (10 * reversed) + (x % 10)
		x /= 10
	}
	return reversed
}

func countDistinctIntegers(nums []int) int {
	visited := make(map[int]bool)
	count := 0
	for _, x := range nums {
		reversed := GetReverseNum(x)
		if visited[x] == false {
			visited[x] = true
			count += 1
		}
		if visited[reversed] == false {
			visited[reversed] = true
			count += 1
		}
	}
	return count
}

File: python/count-number-of-distinct-integers-after-reverse-operations.py
class Solution:
    def get_reverse_num(num: int) -> int:
        reverse_num = 0
        while 0 < num:
            reverse_num = (reverse_num * 10) + (num % 10)
            num //= 10
        return reverse_num

    def countDistinctIntegers(self, nums: List[int]) -> int:
        nums = set(nums)
        new_nums = set()
        count = len(nums)
        for num in nums:
            reverse_num = Solution.get_reverse_num(num)
            if reverse_num not in nums and reverse_num not in new_nums:
                count += 1
                new_nums.add(reverse_num)
        return count
